test/cfa/ntp_conf_test.rb

```diff
@@ -0,0 +1,305 @@
+        expect(ntp.records.first.augeas[:key]).to eq("server[]")
+      end
+    end
+  end
+
+  describe "#records" do
+    let(:content) { ntp_disk_file }
+
+    it "obtains the corrent amount of records" do
+      expect(ntp.records.count).to eq(12)
+    end
+
+    it "obtains a collection of records" do
+      expect(ntp.records).to be_a(CFA::NtpConf::RecordCollection)
+    end
+  end
+end
+
+describe CFA::NtpConf::RecordCollection do
+  subject(:ntp) { CFA::NtpConf.new(file_handler: ntp_file) }
+
+  let(:ntp_file) { CFA::MemoryFile.new(ntp_disk_file) }
+
+  let(:new_record) do
+    record = CFA::NtpConf::ServerRecord.new
+    record.value = "4.pool.ntp.org"
+    record
+  end
+
+  let(:existing_record) { ntp.records.first.dup }
+
+  before do
+    ntp.load
+  end
+
+  context "#add" do
+    context "when does not exist the record to add" do
+      it "adds the record" do
+        ntp.records.add(new_record)
+        expect(ntp.records.include?(new_record)).to be(true)
+      end
+    end
+
+    context "when exists the record to add" do
+      it "adds the record too" do
+        ntp.records.add(existing_record)
+        expect(ntp.records.count(existing_record)).to eq(2)
+      end
+    end
+  end
+
+  context "#delete" do
+    context "when does not exist the record to delete" do
+      it "does anything" do
+        records = ntp.records
+        ntp.records.delete(new_record)
+        expect(ntp.records).to eq(records)
+      end
+    end
+
+    context "when exists the record to delete" do
+      it "deletes the record" do
+        ntp.records.delete(existing_record)
+        expect(ntp.records.include?(existing_record)).to be(false)
+      end
+    end
+  end
+
+  context "#delete_if" do
+    it "deletes all records that satisfy the condition" do
+      ntp.records.delete_if { |record| record.type == "server" }
+      expect(ntp.records.any? { |record| record.type == "server" }).to eq(false)
+    end
+  end
+end
+
+describe CFA::NtpConf::Record do
+  let(:augeas_options) do
+    tree = CFA::AugeasTree.new
+    tree.add("iburst", nil)
+    tree
+  end
+
+  let(:augeas_tree_value) do
+    CFA::AugeasTreeValue.new(augeas_options, "4.pool.ntp.org")
+  end
+
+  let(:augeas_element) do
+    {
+      key:   "server[]",
+      value: augeas_tree_value
+    }
+  end
+
+  subject(:record) { described_class.new_from_augeas(augeas_element) }
+
+  describe ".new_from_augeas" do
+    it "creates a record of the correct class" do
+      expect(record).to be_a(CFA::NtpConf::ServerRecord)
+    end
+
+    it "creates the record with correct data" do
+      expect(record.augeas).to eq(augeas_element)
+    end
+  end
+
+  describe "#value" do
+    it "obtains the value of the record" do
+      expect(record.value).to eq(augeas_element[:value].value)
+    end
+  end
+
+  describe "#value=" do
+    it "sets the value of the record" do
+      value = "1.pool.ntp.org"
+      record.value = value
+      expect(record.value).to eq(value)
+    end
+  end
```

Comment:
A Record is a wrapper of an AugeasElement contained into the AugeasTree of the file. When we change the record, its AugeasElement is modified. For this reason, the changes are always propagated back.